Temporarily revert "fix for nv on powerpc"  upstream commit as it introduced a
-Wint-conversion error with gcc 14 on ppc64le.

See also: https://gitlab.freedesktop.org/xorg/driver/xf86-video-nv/-/commit/1b735e8c9681dcccd54ea0295c4853763dabb8d1

```
nv_driver.c: In function 'NVPreInit':
nv_driver.c:1547:43: error: passing argument 2 of 'vgaHWSetMmioFuncs' makes
pointer from integer without a cast [-Wint-conversion]
 1547 |     vgaHWSetMmioFuncs(VGAHWPTR(pScrn), pNv->IOAddress, 0);
      |                                        ~~~^~~~~~~~~~~
      |                                           |
      |                                           CARD32 {aka unsigned int}
In file included from nv_include.h:52,
                 from nv_driver.c:31:
/usr/include/xorg/vgaHW.h:185:62: note: expected 'CARD8 *' {aka 'unsigned char
*'} but argument is of type 'CARD32' {aka 'unsigned int'}
  185 | extern _X_EXPORT void vgaHWSetMmioFuncs(vgaHWPtr hwp, CARD8 *base, int offset);
      |                                                       ~~~~~~~^~~~
```

--- xf86-video-nv-2.1.23-origin/src/nv_driver.c
+++ xf86-video-nv-2.1.23/src/nv_driver.c
@@ -1543,11 +1543,7 @@
 	xf86FreeInt10(pNv->pInt);
 	return FALSE;
     }
-#ifdef __powerpc__ /* XXX probably MI */
-    vgaHWSetMmioFuncs(VGAHWPTR(pScrn), pNv->IOAddress, 0);
-#else
     vgaHWSetStdFuncs(VGAHWPTR(pScrn));
-#endif
     
     /* We use a programmable clock */
     pScrn->progClock = TRUE;
